var/spack/repos/builtin/packages/py-atlas-building-tools/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAtlasBuildingTools(PythonPackage):
+    """Python tools to build brain region atlases."""
+    homepage = "https://bbpcode.epfl.ch/browse/code/nse/atlas-building-tools/tree/"
+    git      = "ssh://bbpcode.epfl.ch/nse/atlas-building-tools"
+
+    version('0.1.0', commit='4d2ca679e34d08c76f96f15ccd3cdeccd2044696')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('py-cgal-pybind', type=('build', 'run'))
+    depends_on('py-click@3.0:', type=('build', 'run'))
+    depends_on('py-networkx@2.5:', type=('build', 'run'))
+    depends_on('py-nptyping@1.0.1', type=('build', 'run'))
+    depends_on('py-numba@0.48.0:', type=('build', 'run'))
+    depends_on('py-numpy@1.15.0:', type=('build', 'run'))
+    depends_on('py-numpy-quaternion@2017.10.14.11.11.56:2019.12.11.22.25.52', type=('build', 'run'))
```

Comment:
This seems a very specific requirement. Any reason we can't remove at least the upper bound?

Reply:
Regarding numpy-quaternion, there are actually no other choices, unless we force numpy>=1.20.rc1, in which case the versions numpy-quaternion@2021.*.*.*.* can be used.

For the default spack version of numpy, that is 1.19.4, the only compatible versions of numpy-quaternion (always assuming python==3.8.3) are 2017.10.14.11.11.56:2019.12.11.22.25.52. Otherwise you get something like this: https://stackoverflow.com/questions/20518632/importerror-numpy-core-multiarray-failed-to-import.

I'll add a comment.